src/pages/index.js

```diff
@@ -33,155 +36,168 @@ const initialCards = [
-const cardsWrap = document.querySelector(".cards__list");
-const popupImage = document.querySelector(".modal__image");
-const popupCaption = document.querySelector(".modal__caption");
-
-//validation
-const validationSettings = {
-  formSelector: ".modal__form",
-  inputSelector: ".modal__input",
-  submitButtonSelector: ".modal__button",
-  inactiveButtonClass: "modal__button_disabled",
-  inputErrorClass: "modal__input_type_error",
-  errorClass: "modal__error_visible",
-};
+const avatarUrlInput = avatarUpdateForm.querySelector("#avatar-link-input");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "298dccb5-56b1-425a-ba81-960441ee84bf",
+    "Content-Type": "application/json",
+  },
+});
+
+const avatarCardPopUp = new PopUpWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+
+avatarCardPopUp.setEventListeners();
 
 const editFormValidator = new FormValidator(
   validationSettings,
   profileEditForm
 );
 const addFormValidator = new FormValidator(validationSettings, addCardForm);
+const updateAvatarValidator = new FormValidator(
+  validationSettings,
+  avatarUpdateForm
+);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
 editFormValidator.disableButton();
 addFormValidator.disableButton();
+updateAvatarValidator.enableValidation();
+updateAvatarValidator.disableButton();
 
 /* const closeModal = (modalWindow) => {
   modalWindow.classList.remove("modal_open");
   document.removeEventListener("keydown", handleEscapeKey);
 }; */
 
 function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleCardClick);
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    handleCardClick,
+    deleteCardSubmit,
+    handleLikeClick
+  );
   return cardElement.getView();
 }
 
-/* con = (data, wrap) => {
-  const card = createCard(data);
-  wrap.prepend(card);
-}; */
-
 const profileName = document.querySelector(".profile__name");
 const profileDescription = document.querySelector(".profile__description");
-
+const avatar = document.querySelector(".profile__image");
 const profileNameInput = document.querySelector("#profile-name-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 
-const userData = new UserInfo(profileName, profileDescription);
+function handleProfileEditSubmit(userProfile) {
+  popupWithEditForm.renderLoading(true);
+
+  api
+    .updateUserInfo(userProfile.name, userProfile.link)
+    .then(() => {
+      userData.setUserInfo(userProfile.name, userProfile.link);
+    })
+    .then(() => {
+      editFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      popupWithEditForm.renderLoading(false);
+    });
 
-function handleProfileEditSubmit() {
-  userData.setUserInfo(profileNameInput.value, profileDescriptionInput.value);
   popupWithEditForm.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


**Please, correct it everywhere**